molgenis-core-ui/src/main/java/org/molgenis/core/ui/menumanager/MenuManagerServiceImpl.java

```diff
@@ -3,114 +3,41 @@
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.Collectors.toList;
 
-import com.google.gson.GsonBuilder;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import org.molgenis.core.ui.MenuType;
-import org.molgenis.core.ui.Molgenis;
-import org.molgenis.core.ui.PluginType;
-import org.molgenis.core.ui.XmlMolgenisUiLoader;
-import org.molgenis.core.ui.menu.Menu;
-import org.molgenis.core.ui.menu.MenuItem;
-import org.molgenis.core.ui.menu.MenuItemType;
-import org.molgenis.core.ui.menu.MenuReaderService;
+import com.google.gson.Gson;
 import org.molgenis.data.DataService;
 import org.molgenis.data.plugin.model.Plugin;
 import org.molgenis.data.plugin.model.PluginMetadata;
 import org.molgenis.security.core.runas.RunAsSystem;
 import org.molgenis.settings.AppSettings;
+import org.molgenis.web.menu.model.Menu;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
 @Component
 public class MenuManagerServiceImpl implements MenuManagerService {
-  private final MenuReaderService menuReaderService;
   private final AppSettings appSettings;
   private final DataService dataService;
+  private final Gson gson;
 
-  public MenuManagerServiceImpl(
-      MenuReaderService menuReaderService, AppSettings appSettings, DataService dataService) {
-    this.menuReaderService = requireNonNull(menuReaderService);
+  public MenuManagerServiceImpl(AppSettings appSettings, DataService dataService, Gson gson) {
     this.appSettings = requireNonNull(appSettings);
     this.dataService = requireNonNull(dataService);
-  }
-
-  @Override
-  @PreAuthorize("hasAnyRole('ROLE_SYSTEM, ROLE_SU, ROLE_PLUGIN_READ_menumanager')")
-  @Transactional(readOnly = true)
-  public Menu getMenu() {
-    return menuReaderService.getMenu();
+    this.gson = requireNonNull(gson);
   }
 
   @Override
   @RunAsSystem
-  @PreAuthorize("hasAnyRole('ROLE_SYSTEM, ROLE_SU, ROLE_PLUGIN_READ_menumanager')")
```

Comment:
I was surprised to find the role here, since the construction ROLE_PLUGIN_PERMISSION_pluginID was removed when we switched to plugin permissions using ACLs. The role does not exist anymore so this will not affect users. Also, non-su people should not be editing the menu because they only see part of it and saving the menu will remove the bits they have no permission to see.